assets/src/editor/events/dates-and-times/editor-date/actions-menu/editor-date-entity-actions-menu.js

```diff
@@ -3,196 +3,184 @@
-				id={ `edit-date-${ dateEntity.id }` }
-				htmlClass="edit-date"
-				dashicon="edit"
-				onClick={ this.props.toggleEditor }
-			/>
-		);
-	};
+	const editDateMenuItem = useCallback(
+		() => {
+			return (
+				<IconMenuItem
+					index={ 1 }
+					tooltip={ __( 'edit date', 'event_espresso' ) }
+					id={ `edit-date-${ dateEntity.id }` }
+					htmlClass="edit-date"
+					dashicon="edit"
+					onClick={ toggleDateEditor }
+				/>
+			);
+		},
+		[ dateEntity, toggleDateEditor ]
+	);
 
 	/**
 	 * @function
 	 * @param {Object} dateEntity JSON object defining the Event Date
 	 * @param {Array} ticketEntities    Tickets for Event Date
 	 * @return {IconMenuItem}    View Tickets for Event Date IconMenuItem
 	 */
-	viewTicketsMenuItem = ( dateEntity, ticketEntities ) => {
-		const tooltip = isEmpty( ticketEntities ) ?
-			__(
-				'warning! no assigned tickets - click to fix',
-				'event_espresso'
-			) :
-			__( 'assign tickets', 'event_espresso' );
-		return (
-			<IconMenuItem
-				index={ 2 }
-				tooltip={ tooltip }
-				id={ `view-tickets-date-${ dateEntity.id }` }
-				htmlClass="view-tickets-date"
-				dashicon="tickets-alt"
-				onClick={ this.props.toggleTicketAssignments }
-				itemCount={ ticketEntities.length }
-			/>
-		);
-	};
+	const viewTicketsMenuItem = useCallback(
+		() => {
+			const tooltip = isEmpty( ticketEntities ) ?
+				__(
+					'warning! no assigned tickets - click to fix',
+					'event_espresso'
+				) :
+				__( 'assign tickets', 'event_espresso' );
+			return (
+				<IconMenuItem
+					index={ 2 }
+					tooltip={ tooltip }
+					id={ `view-tickets-date-${ dateEntity.id }` }
+					htmlClass="view-tickets-date"
+					dashicon="tickets-alt"
+					onClick={ toggleTicketAssignments }
+					itemCount={ ticketEntities.length }
+				/>
+			);
+		},
+		[ dateEntity, ticketEntities, toggleTicketAssignments ]
+	);
 
 	/**
 	 * @function
 	 * @param {Object} dateEntity    	 JSON object defining the Event Date
 	 * @param {Array} ticketEntities    Tickets for Event Date
 	 * @return {Array}    				 Array of IconMenuItem objects
 	 */
-	getSidebarMenuItems = ( dateEntity, ticketEntities ) => {
-		const sidebarMenuItems = [];
-		sidebarMenuItems.push(
-			this.mainDropDownMenu( dateEntity )
-		);
-		sidebarMenuItems.push( this.editDateMenuItem( dateEntity ) );
-		sidebarMenuItems.push(
-			this.viewTicketsMenuItem( dateEntity, ticketEntities )
-		);
-		return applyFilters(
-			'FHEE__EditorDateEntityActionsMenu__SidebarMenuItems',
-			sidebarMenuItems,
-			dateEntity
-		);
-	};
+	const getSidebarMenuItems = useCallback(
+		() => {
+			const sidebarMenuItems = [];
+			sidebarMenuItems.push(
+				mainDropDownMenu()
+			);
+			sidebarMenuItems.push( editDateMenuItem() );
+			sidebarMenuItems.push(
+				viewTicketsMenuItem()
+			);
+			return applyFilters(
+				'FHEE__EditorDateEntityActionsMenu__SidebarMenuItems',
+				sidebarMenuItems,
+				dateEntity
+			);
+		},
+		[ mainDropDownMenu, editDateMenuItem, viewTicketsMenuItem ]
```

Comment:
this is currently using `applyFilters()` and I know you have suggested changing this to utilize slot fill instead,
but regardless of which method is employed, how does it affect the dependencies for `useCallback()` ?
What if the listed dependencies don't change but the conditions in the logic filtering `sidebarMenuItems` do, and it needs to change what is returned?
For example, what if REM changes the state for what it is injecting here while the listed dependencies remain the same?

I'm not expecting you to have an answer and am content with having to just wait and see what happens.

Reply:
Your observations/questions are good.  I had the same thoughts when I was implementing this and forgot to leave inline comments about it.  It's a good example why implementing filters in reactive components is difficult because there is no way to reliably handle scenarios where things hooking into a filter may  introduce other dependencies in react hooks etc.  I decided to leave as is for now because with the refactor to slot/fill it will become a moot issue.